File: javascript/rna-transcription/rna-transcription.js
function DnaTranscriber() {
	this.toRna = function(input) {
		this.value = input;
		let result = [];
		for (var i = 0; i!==this.value.length; i++) {
			switch (this.value[i]) {
				case 'C':
					result = result + 'G';
					break;
				case 'G':
					result = result + 'C';
					break;
				case 'T':
					result = result + 'A';
					break;
				case 'A':
					result = result + 'U';
					break;
				default:
					throw new Error('Invalid input');
			}
		}
		return result;
	}
};

module.exports = DnaTranscriber;

// ATCA 

File: java/rna-transcription/src/main/java/RnaTranscription.java
class RnaTranscription {

    String transcribe(String dnaStrand) {
    	String result = "";
        for (int i=0;i < dnaStrand.length();i++) {
        	switch (dnaStrand.charAt(i)) {
        		case 'G': result += 'C';
        				  break;
        		case 'C': result += 'G';
        				  break;
        		case 'T': result += 'A';
        				  break;
        		case 'A': result += 'U';
        				  break;
        	    default: throw new IllegalArgumentException("Invalid input");
        	}
        }
        return result;
    }

}


/*
* `G` -> `C`
* `C` -> `G`
* `T` -> `A`
* `A` -> `U`
*/